docs/react/use-with-create-react-app-ts.en-US.md

```diff
@@ -0,0 +1,236 @@
+---
+order: 4
+title: Use in create-react-app-ts
+---
+
+[create-react-app](https://github.com/facebookincubator/create-react-app) is one of best React application development tool, we are going to use `antd` within it and modify the webpack config for some customized needs.
+
+---
+
+## Install and Initialization
+
+We need to install `create-react-app` first, you may need install [yarn](https://github.com/yarnpkg/yarn/) too.
+
+Additionally, install [react-scripts-ts](https://github.com/wmonk/create-react-app-typescript)
+
+```bash
+$ npm install -g create-react-app yarn react-scripts-ts
+```
+
+Create a new project named `antd-demo-ts`.
+
+```bash
+$ create-react-app antd-demo-ts --scripts-version=react-scripts-ts
+```
+
+The tool will create and initialize environment and dependencies automatically,
+please try config your proxy setting or use other npm registry if any network errors happen during it.
+
+Then we go inside `antd-demo-ts` and start it.
+
+```bash
+$ cd antd-demo-ts
+$ yarn start
+```
+
+Open browser at http://localhost:3000/, it renders a header saying "Welcome to React" on the page.
+
+## Import antd
+
+Default directory structure below. 
+
+```
+├──.gitignore
+│──package.json
+│──README.md
+│──tsconfig.json
+│──tslint.json
+│──yarn.lock
+├── public
+│──├── favicon.ico
+│──└── index.html
+├──src
+│──└──App.css
+│──└──  App.test.tsx
+│──└──  App.tsx
+│──└──  index.css
+│──└──  index.tsx
+│──└──  logo.svg
+```
+
+Now we install `antd` from yarn or npm.
+
+```bash
+$ yarn add antd --save
+```
+
+Modify `src/App.tsx`, import Button component from `antd`.
+
+```jsx
+import * as React from 'react';
+import './App.css';
+import { Button } from 'antd';
+
+class App extends React.Component<null, null> {
+  render() {
+    return (
+      <div className="App">
+        <p>
+          <Button type="primary">Hello World</Button>
+        </p>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+Add `antd/dist/antd.css` at the top of `src/App.css`.
+
+```css
+@import '~antd/dist/antd.css';
+
+.App {
+  text-align: center;
+}
+
+...
+```
+
+Ok, you now see a blue primary button displaying in page now, next you can choose any components of `antd` to develop your application. Visit other workflow of `create-react-app` at its [User Guide ](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md).
+
```

Comment:
All previous steps work perfectly, but at this step, I run into this error : 
```
C:\node-projects\antd-demo-ts\node_modules\antd\lib\upload\interface.d.ts
(2,8): error TS1192: Module '"C:/node-projects/antd-demo-ts/node_modules/@types/react/index"' has no default export.
```

Reply:
I've got a different error (after running up to this point):
```
/home/user/projects/antd-demo-ts/node_modules/antd/lib/upload/interface.d.ts
(2,8): error TS1192: Module '"/home/user/projects/antd-demo-ts/node_modules/@types/react/index"' has no default export.
```
This is fixed by adding `"allowSyntheticDefaultImports": true` to tsconfig

Reply:
yes the issue I mentionned is fixed, but `allowSyntheticDefaultImports` is still required, I've added it at https://github.com/abenhamdine/ant-demo-ts-cra/blob/master/README.md#step-7--edit-tsconfigjson